src/main/java/core/basesyntax/service/RegistrationServiceImpl.java

```diff
@@ -2,13 +2,48 @@
 
 import core.basesyntax.dao.StorageDao;
 import core.basesyntax.dao.StorageDaoImpl;
+import core.basesyntax.db.Storage;
+import core.basesyntax.exception.UserNotAllowed;
 import core.basesyntax.model.User;
+import java.util.HashSet;
+import java.util.Set;
 
 public class RegistrationServiceImpl implements RegistrationService {
+    private static final int MINIMAL_LOGIN_LENGTH = 6;
+    private static final int MINIMAL_USER_AGE = 18;
+    private static final int MINIMAL_PASSWORD_LENGTH = 6;
     private final StorageDao storageDao = new StorageDaoImpl();
+    private Set<String> loginSet = new HashSet<>();
```

Comment:
The `loginSet` is being used to track unique logins, but it's being initialized every time `register` is called. Consider moving the initialization to the constructor or using a different approach to ensure uniqueness, such as checking directly against the `Storage.people`.